pkg/rootless/rootless_linux.c

```diff
@@ -506,15 +506,15 @@ create_pause_process (const char *pause_pid_file_path, char **argv)
   if (pid)
     {
       char b;
-      int r;
+      int r, r2;
 
       close (p[1]);
       /* Block until we write the pid file.  */
       r = TEMP_FAILURE_RETRY (read (p[0], &b, 1));
       close (p[0]);
 
-      r = reexec_in_user_namespace_wait (pid, 0);
-      if (r != 0)
+      r2 = reexec_in_user_namespace_wait (pid, 0);
```

Comment:
This looks very different in the previous code reexec_in_user_namespace_wait was expected to succeed if it returned a 1, no you are failing it if returns a 1?

@giuseppe PTAL

Reply:
this looks indeed like a regression that was introduced with 99258703a7c593c2717438928cd9c9052e50c26b as it overwrites the value we care about.